state/state.go

```diff
@@ -752,5 +752,8 @@ func (st *state) publishEvent(msg any) {
 		return
 	}
 
-	st.eventCh <- msg
+	select {
```

Comment:
Why do we need select here?

Reply:
> Why do we need select here?

Select prevent panic on send message on closed channel.